### PR TITLE
docs: improve api docs

### PR DESCRIPTION
**Pull request by wizardofhogwarts**

### CODE REVIEW

File: src/content/api/compilation-hooks.md [hunk 1]
```diff
@@ -15,7 +16,7 @@ During the compilation phase, modules are loaded, sealed, optimized, chunked,
 hashed and restored.
 
 The `Compilation` class also extends `Tapable` and provides the following
-lifecycle hooks. They can be tapped the same way as compiler hooks:
+life-cycle hooks. They can be tapped the same way as compiler hooks:
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md]: I think this can also be `life cycle` and `lifecycle`. Why updating it now?

File: src/content/api/loaders.md [hunk 2]
```diff
@@ -185,12 +186,12 @@ require('./loader1?xyz!loader2!./resource?rrr');
 
 ### `this.version`
 
-__Loader API version.__ Currently `2`. This is useful for providing backwards compatibility. Using the version you can specify custom logic or fallbacks for breaking changes.
+__Loader API version.__ Currently `2`. This is useful for providing backward compatibility. Using the version you can specify custom logic or fallbacks for breaking changes.
```
Comment (EugeneHlushko) [inline src/content/api/loaders.md]: Why is this necessary?

File: src/content/api/loaders.md [hunk 3]
```diff
@@ -337,7 +338,7 @@ T> Loaders were originally designed to also work as Babel transforms. Therefore
 
 ### `this.sourceMap`
 
-Should a source map be generated. Since generating source maps can be an expensive task, you should check if source maps are actually requested.
+Should a source map be generated? Since generating source maps can be an expensive task, you should check if source maps are actually requested.
```
Comment (EugeneHlushko) [inline src/content/api/loaders.md]: This might be better imo
```suggestion
Tells if source map should be generated. Since generating source maps can be an expensive task, you should check if source maps are actually requested.
```

File: src/content/api/loaders.md [hunk 4]
```diff
@@ -480,7 +481,7 @@ A boolean flag. It is set when in debug mode.
 
 ### `this.minimize`
 
-Should the result be minimized.
+Should the result be minimized?
```
Comment (EugeneHlushko) [inline src/content/api/loaders.md]:
```suggestion
Tells if result should be minimized.
```

File: src/content/api/module-methods.md [hunk 5]
```diff
@@ -167,7 +168,7 @@ W> Module ID is a number in webpack (in contrast to NodeJS where it is a string
 
 ### `require.cache`
 
-Multiple requires to the same module result in only one module execution and only one export. Therefore a cache in the runtime exists. Removing values from this cache cause new module execution and a new export.
+Multiple requires in the same module result in only one module execution and only one export. Therefore a cache in the runtime exists. Removing values from this cache cause new module execution and a new export.
```
Comment (EugeneHlushko) [inline src/content/api/module-methods.md]: I think this actually ment: "multiple requires of a specific module", so please revert

Comment (EugeneHlushko) [reply src/content/api/module-methods.md]:
```suggestion
Multiple requires of the same module result in only one module execution and only one export. Therefore a cache in the runtime exists. Removing values from this cache cause new module execution and a new export.
```

Comment (EugeneHlushko) [reply src/content/api/module-methods.md]: Or change to of as suggested

File: src/content/api/stats.mdx [hunk 6]
```diff
@@ -176,7 +177,7 @@ The `errors` and `warnings` properties each contain a list of strings. Each stri
 ``` bash
 ../cases/parsing/browserify/index.js
 Critical dependencies:
-2:114-121 This seem to be a pre-built javascript file. Even while this is possible, it's not recommended. Try to require to original source to get better results.
+2:114-121 This seems to be a pre-built javascript file. Even while this is possible, it's not recommended. Try to require the original source to get better results.
```
Comment (EugeneHlushko) [inline src/content/api/stats.mdx]: Please revert, this is example output of an error which is not controlled by the docs website.

File: src/content/api/compilation-hooks.md (hunk 1 above)
Comment (wizardofhogwarts) [reply src/content/api/compilation-hooks.md]: I just felt that life-cycle is better.

File: src/content/api/loaders.md (hunk 2 above)
Comment (wizardofhogwarts) [reply src/content/api/loaders.md]: In this context, backward makes more sense.

File: src/content/api/stats.mdx (hunk 6 above)
Comment (wizardofhogwarts) [reply src/content/api/stats.mdx]: oh! I hadn't noticed that.

File: src/content/api/module-methods.md (hunk 5 above)
Comment (wizardofhogwarts) [reply src/content/api/module-methods.md]: @EugeneHlushko yeah, it looks better with `of`.

File: src/content/api/stats.mdx (hunk 6 above)
Comment (wizardofhogwarts) [reply src/content/api/stats.mdx]: @EugeneHlushko I just committed to change it back to seem

File: src/content/api/loaders.md (hunk 2 above)
Comment (EugeneHlushko) [reply src/content/api/loaders.md]: I have read an article on it, it seems both versions are correct. Unless there is more reading to it?

File: src/content/api/compilation-hooks.md (hunk 1 above)
Comment (EugeneHlushko) [reply src/content/api/compilation-hooks.md]: Same goes for life-cycle, i had found few resources on the internet which ensure me that all three versions can be used, and be correct. Unless this is wrong/misused word i would suggest reverting the changes to not change for the sake of change

Comment (wizardofhogwarts) [reply src/content/api/compilation-hooks.md]: Ok i'll revert it.

File: src/content/api/loaders.md (hunk 2 above)
Comment (wizardofhogwarts) [reply src/content/api/loaders.md]: Well, what I just figured out is that `backward` is American variant. So, changing it won't create extraordinary improvement. So, reverting it.

File: src/content/api/compilation-hooks.md [hunk 7]
```diff
@@ -541,6 +541,8 @@ compilation.hooks.afterOptimizeChunkAssets.tap('MyPlugin', chunks => {
 ```
 
 
+
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md]: remove this extra 2 blank lines please

File: src/content/api/stats.mdx [hunk 8]
```diff
@@ -176,7 +177,7 @@ The `errors` and `warnings` properties each contain a list of strings. Each stri
 ``` bash
 ../cases/parsing/browserify/index.js
 Critical dependencies:
-2:114-121 This seem to be a pre-built javascript file. Even while this is possible, it's not recommended. Try to require to original source to get better results.
+2:114-121 This seem to be a pre-built javascript file. Even while this is possible, it's not recommended. Try to require the original source to get better results.
```
Comment (EugeneHlushko) [inline src/content/api/stats.mdx]: this looks like output from other software, please revert if so

Comment (wizardofhogwarts) [reply src/content/api/stats.mdx]: @EugeneHlushko to two times was lookin awkward that's why i did this. What u say?

File: src/content/api/compilation-hooks.md (hunk 7 above)
Comment (wizardofhogwarts) [reply src/content/api/compilation-hooks.md]: @EugeneHlushko Oh! my bad. Reverted. Thanks.

File: src/content/api/stats.mdx (hunk 8 above)
Comment (wizardofhogwarts) [reply src/content/api/stats.mdx]: @EugeneHlushko just reverted the output to it's original state

**Comment by wizardofhogwarts** [comment]
@EugeneHlushko Now check the suggested changes. They are implemented as per your suggestions.

File: src/content/api/stats.mdx [hunk 9]
```diff
@@ -5,6 +5,7 @@ contributors:
   - skipjack
   - franjohn21
   - byzyk
+  - wizardofhogwarts
```
Comment (EugeneHlushko) [inline src/content/api/stats.mdx]: This isn't necessary yet :)

Comment (wizardofhogwarts) [reply src/content/api/stats.mdx]: ohh yeah!! @EugeneHlushko thanks for figuring out that.

**Comment by EugeneHlushko** [comment]
Lets see the tests pass after updating to latest master

**Comment by EugeneHlushko** [comment]
Thanks!
